dotnet-authserver/src/TeacherIdentity.AuthServer/RequestClientIpProvider.cs

```diff
@@ -0,0 +1,17 @@
+namespace TeacherIdentity.AuthServer;
+
+public class RequestClientIpProvider : IRequestClientIpProvider
+{
+    private IHttpContextAccessor _httpContextAccessor { get; }
+
+    public RequestClientIpProvider(IHttpContextAccessor httpContextAccessor)
+    {
+        _httpContextAccessor = httpContextAccessor;
+    }
+
+    public string GetClientIPAddress()
+    {
+        var ip = _httpContextAccessor.HttpContext?.Connection.RemoteIpAddress?.ToString();
+        return ip!;
```

Comment:
The signature doesn't support returning `null` so this needs to throw instead:
```suggestion
        var httpContext = _httpContextAccessor.HttpContext ?? throw new InvalidOperationException("No HttpContext.");
        var ip = httpContext.Connection.RemoteIpAddress?.ToString() ?? throw new Exception("Connection has no RemoteIpAddress");
        return ip!;
```